### PR TITLE
Updated to not use the deprecated mach_port_destroy

### PR DESCRIPTION
**Pull request by LowAmmo**
Cocoapods
* Updated to set iOS 12.0 as the minimum supported version

This PR is to address Issue #29

### CODE REVIEW

**Comment by LowAmmo** [comment]
@mattgallagher - Put this PR together, also, to eliminate all build warnings being seen with Xcode 14 (in conjunction with mattgallagher/CwlCatchException#8 ).

Same as that PR, let me know if there's anything you want me to change (including reverting the minimum iOS version back to 9.0).

-Thanks!

File: Sources/CwlPreconditionTesting/CwlCatchBadInstruction.swift [hunk 1]
```diff
@@ -171,7 +171,10 @@ public func catchBadInstruction(in block: @escaping () -> Void) -> BadInstructio
 		}
 		defer {
 			// 7. Cleanup the mach port
-			mach_port_destroy(mach_task_self_, context.currentExceptionPort)
+            // When the reference count for the right goes down to 0, it triggers the right to be deallocated
+            mach_port_mod_refs(mach_task_self_, context.currentExceptionPort, MACH_PORT_RIGHT_RECEIVE, -1)
+            // All rights deallocated, can deallocate the name/port
+            mach_port_deallocate(mach_task_self_, context.currentExceptionPort)
```
Comment (mattgallagher) [inline Sources/CwlPreconditionTesting/CwlCatchBadInstruction.swift:177]: I'm curious about this change as I don't see any warnings or documentation about the deprecation of mach_port_destroy. Can you tell me where this came from?

**Review by mattgallagher — APPROVED** [review]
I'm going to merge the podspec changes but I'm going to hold off on the mach_port_destroy changes until I can confirm what this is supposed to do.

File: Sources/CwlPreconditionTesting/CwlCatchBadInstruction.swift (hunk 1 above)
Comment (mattgallagher) [reply Sources/CwlPreconditionTesting/CwlCatchBadInstruction.swift:177]: Aah, nevermind... I've found it:

```
@available(iOS, introduced: 2.0, deprecated: 15.0, message: "Inherently unsafe API: instead manage rights with mach_port_destruct(), mach_port_deallocate() or mach_port_mod_refs()")
public func mach_port_destroy(_ task: ipc_space_t, _ name: mach_port_name_t) -> kern_return_t
```

I'll restore this change before creating the tag.

**Comment by LowAmmo** [comment]
Thanks, @mattgallagher !

Yeah, took me guessing and trying things out to get what I believe is the correct way - guessing you arrived at the same conclusion.

I'll keep an eye for everything to get updated in the cocoapods spec repo!

-Thanks!

**Pull request merged**